### PR TITLE
chore: update checks for modfiles

### PR DESCRIPTION
**Pull request by Racer159**
## Description

Adds some guardrails to release while cross-team talks happen for release engineering of `pkg`.

### CODE REVIEW

File: internal/release/main.go [hunk 1]
```diff
@@ -29,29 +30,55 @@ func getProjectPath() (string, error) {
...
+
+	// Strip the /vX from the end of any mod paths
+	actualModPathSections := strings.Split(actualModPath, "/")
+	if len(actualModPathSections) == 5 {
+		actualModPath = strings.Join(actualModPathSections[:4], "/")
+		actualMajorVersion, err = strconv.ParseInt(strings.TrimPrefix(actualModPathSections[4], "v"), 10, 0)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (Noxsios) [inline internal/release/main.go:62]: this does assume that we will never do non top-level modules,

ie: `github.com/defenseunicorns/pkg/auth/retry` for example

Comment (Racer159) [reply internal/release/main.go:62]: correct

**Comment by marshall007** [comment]
@Racer159 in the future we might want to consider `goreleaser`: https://goreleaser.com/ci/actions/

It supports building [multiple modules](https://goreleaser.com/customization/builds/) in a single pass, handles [generation of changelog](https://goreleaser.com/customization/changelog/) (from GitHub PRs or git commits), [publishing GitHub releases](https://goreleaser.com/customization/release/#github), [nightlies](https://goreleaser.com/customization/nightlies/), etc.

If there are projects where we are publishing prebuilt binaries [it can sign them with cosign](https://goreleaser.com/customization/sign/#with-cosign) too.

**Comment by oates** [comment]
adding a comment

**Review by Noxsios — APPROVED** [review]

**Pull request merged**